src/App.js

```diff
@@ -17,26 +17,29 @@ const requireSignIn = generateRequireSignInWrapper({
 
 class App extends Component {
   state = {
-    paths: []
+    paths: [],
+    country: ''
   }
 
   async componentWillMount() {
     let categoryPaths = await getCategoryPaths()
     this.setState({ paths: categoryPaths })
   }
+
   render() {
     return (
       <>
-        <HeaderMain />
-        <Switch>
-          <Route exact path='/' render={() => (<Redirect to="/news" component={ArticlesByCategory} activeItem={'news'} />)}></Route>
-          <Route exact path={this.state.paths} component={ArticlesByCategory}></Route>
-          <Route exact path='/write-article' component={requireSignIn(WriteArticle)}></Route>
-          <Route exact path='/review-articles' component={requireSignIn(ListOfUnpublishedArticles)}></Route>
-          <Route exact path='/full-article' component={requireSignIn(FullArticle)}></Route>
-          <Route exact path='/login' component={LoginForm}></Route>
-          <Route exact path='/signup' component={SignUpForm}></Route>
-        </Switch>
+        <HeaderMain
+          handleChangeCountry={this.handleChangeCountry} />
+          <Switch>
+            <Route exact path='/' render={() => (<Redirect to="/news" component={ArticlesByCategory} activeItem={'news'}/>)}></Route>
+            <Route exact path={this.state.paths} render={(props) => (<ArticlesByCategory {...props}/>)}></Route>
```

Comment:
Just out of curiosity: Is this line of passing props to ArticlesByCategory still 'necessary' with Redux?

Reply:
True also my friend. Not needed. Thanks.